streams/src/test/java/org/apache/kafka/test/ProcessorTopologyTestDriver.java

```diff
@@ -187,7 +188,7 @@ public ProcessorTopologyTestDriver(StreamsConfig config, TopologyBuilder builder
 
         // Set up all of the topic+partition information and subscribe the consumer to each ...
         for (String topic : topology.sourceTopics()) {
-            TopicPartition tp = new TopicPartition(topic, 1);
+            TopicPartition tp = new TopicPartition(topic, id.partition);
```

Comment:
The problem is that `ProcessorStateManager.storeChangelogTopic(String applicationId, String storeName)` is a static method that returns `<applicationId>-<storeName>-changelog`.

Sometimes, a store does not have an internal changelog topic, e.g. when creating a KTable for a topic. `topology.storeToChangelogTopic()` returns the actual changelog topics.